File: app/src/main/java/com/mohsenoid/rickandmorty/RickAndMortyApplication.java
package com.mohsenoid.rickandmorty;

import android.app.Application;

import com.mohsenoid.rickandmorty.injection.DependenciesProvider;

public class RickAndMortyApplication extends Application {

    private DependenciesProvider dependenciesProvider;

    public DependenciesProvider getDependenciesProvider() {
        return dependenciesProvider;
    }

    @Override
    public void onCreate() {
        super.onCreate();

        dependenciesProvider = new DependenciesProvider(this);
    }
}

File: app/src/main/java/com/mohsenoid/rickandmorty/RickAndMortyApplication.kt
package com.mohsenoid.rickandmorty

import android.app.Application
import com.mohsenoid.rickandmorty.data.dataModule
import com.mohsenoid.rickandmorty.domain.domainModule
import com.mohsenoid.rickandmorty.ui.uiModule
import org.koin.android.ext.koin.androidContext
import org.koin.android.ext.koin.androidLogger
import org.koin.core.context.startKoin

class RickAndMortyApplication : Application() {
    override fun onCreate() {
        super.onCreate()
        startKoin {
            androidLogger()
            androidContext(this@RickAndMortyApplication)
            modules(appModule, dataModule, domainModule, uiModule)
        }
    }
}
